Sources/Purchasing/StoreKitAbstractions/SK2StoreTransaction.swift

```diff
@@ -16,14 +16,17 @@ import StoreKit
 @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
 internal struct SK2StoreTransaction: StoreTransactionType {
 
-    init(sk2Transaction: SK2Transaction, jwsRepresentation: String) {
+    init(sk2Transaction: SK2Transaction, jwsRepresentation: String, environment: StoreEnvironment? = nil) {
         self.underlyingSK2Transaction = sk2Transaction
 
         self.productIdentifier = sk2Transaction.productID
         self.purchaseDate = sk2Transaction.purchaseDate
         self.transactionIdentifier = String(sk2Transaction.id)
         self.quantity = sk2Transaction.purchasedQuantity
         self.jwsRepresentation = jwsRepresentation
+        self.environment = environment ?? StoreEnvironment(
+            rawValue: sk2Transaction.environmentStringRepresentation.lowercased()
+        )
```

Comment:
`sk2Transaction ` is a `StoreKit.Transaction` and its `environment` field is immutable. It will always be `Xcode` in tests, but we want to be able to test the behavior when the environment is `Sandbox`/`Production`.

Reply:
> we want to be able to test the behavior when the environment is Sandbox/Production.

That's something we can override with `MockTransaction` though, right?

Reply:
Not in tests where we're using a transaction created via StoreKit Testing in a `StoreKitConfigTestCase`, unless I'm missing something?